lms/djangoapps/course_api/views.py

```diff
@@ -193,17 +195,34 @@ class CourseListView(DeveloperErrorViewMixin, ListAPIView):
     pagination_class = NamespacedPageNumberPagination
     serializer_class = CourseSerializer
 
+    # Return all the results, 10K is the maximum allowed value for ElasticSearch.
+    # We should use 0 after upgrading to 1.1+:
+    #   - https://github.com/elastic/elasticsearch/commit/8b0a863d427b4ebcbcfb1dcd69c996c52e7ae05e
+    results_size_infinity = 10000
+
     def get_queryset(self):
         """
         Return a list of courses visible to the user.
         """
-        form = CourseListGetForm(self.request.query_params, initial={'requesting_user': self.request.user})
+        form = CourseListGetAndSearchForm(self.request.query_params, initial={'requesting_user': self.request.user})
         if not form.is_valid():
             raise ValidationError(form.errors)
 
-        return list_courses(
+        courses_db = list_courses(
             self.request,
             form.cleaned_data['username'],
             org=form.cleaned_data['org'],
             filter_=form.cleaned_data['filter_'],
         )
+
+        courses_search = search.api.course_discovery_search(
```

Comment:
I am not seeing where `search` is coming from in this PR.  Is it part of the file already?  I am looking through the code locally and also not seeing it.

Reply:
@staubina I think you're referring to this repo:

 - https://github.com/edx/edx-search